docs/source/tutorial/local-state.mdx

```diff
@@ -567,23 +526,84 @@ const ActionButton: React.FC<ActionButtonProps> = ({ isBooked, id, isInCart }) =
         onClick={() => mutate()}
         data-testid={'action-button'}
       >
-        {isBooked
-          ? 'Cancel This Trip'
-          : isInCart
-            ? 'Remove from Cart'
-            : 'Add to Cart'}
+        Cancel This Trip
+      </Button>
+    </div>
+  );
+};
+
+const ToggleTripButton: React.FC<ActionButtonProps> = ({ id }) => {
+  const cartItems = cartItemsVar();
+  const isInCart = id ? cartItems.includes(id) : false;
+  const [, toggleTrip] = useState(isInCart);
+  return (
+    <div>
+      <Button
+        onClick={() => {
+          if (id) {
+            cartItemsVar(
+              isInCart
+                ? cartItems.filter((i) => i !== id)
+                : [...cartItems, id]
+            );
+            // Trigger a re-render to pick up the `cartItemsVar` changes.
+            toggleTrip(!isInCart);
+          }
+        }}
+        data-testid={'action-button'}
+      >
+        {isInCart ? 'Remove from Cart' : 'Add to Cart'}
```

Comment:
```suggestion
const ToggleTripButton: React.FC<ActionButtonProps> = ({ id }) => {
  const cartItems = useReactiveVar(cartItemsVar);
  const isInCart = id ? cartItems.includes(id) : false;
  return (
    <div>
      <Button
        onClick={() => {
          if (id) {
            cartItemsVar(
              isInCart
                ? cartItems.filter(itemId => itemId !== id)
                : [...cartItems, id]
            );
          }
        }}
        data-testid={'action-button'}
      >
        {isInCart ? 'Remove from Cart' : 'Add to Cart'}
```
This can be a bit simpler (no need for `useState`) if you use [`useReactiveVar`](https://github.com/apollographql/apollo-client/pull/6867).

As a side note, it might be a good idea to reevaluate `cartItemsVar()` inside the `onClick` handler, so you aren't modifying a stale value. However, in this example, I don't think that causes any problems in practice.